homeassistant/components/homeassistant/__init__.py

```diff
@@ -203,4 +207,23 @@ async def async_set_location(call):
         vol.Schema({ATTR_LATITUDE: cv.latitude, ATTR_LONGITUDE: cv.longitude}),
     )
 
+    async def async_handle_reload_config_entry(call):
+        """Service handler for reloading a config entry."""
+        reload_entries = await async_extract_config_entry_ids(hass, call)
+        if not reload_entries:
+            raise ValueError("There were no matching config entries to reload")
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_reload(config_entry_id)
+                for config_entry_id in reload_entries
+            ]
+        )
+
+    hass.helpers.service.async_register_admin_service(
+        ha.DOMAIN,
+        SERVICE_RELOAD_CONFIG_ENTRY,
+        async_handle_reload_config_entry,
+        schema=cv.make_entity_service_schema({}),
```

Comment:
Since it's such an advanced functionality, I don't think it's bad to just accept a `config_entry_id`.

Reply:
Will add that as an option and mark it as advanced in services.y'all